### PR TITLE
Add option to build a package for a specific architecture

### PR DESCRIPTION
**Pull request by dallasd1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This change allows you to build a package for either arm64 for x86_64 by appending `TARGET_ARCH` to the `build-packages` make command. This change combined with the cross-toolchain allows developers to cross-build RPM packages.

###### Change Log  

- Add argument TARGET_ARCH to the build-packages command

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Test Methodology

- Local build

### CODE REVIEW

**Comment by dmcilvaney** [comment]
The go auto formatter looks unhappy, you can invoke all our tidy scripts via `cd ./toolkit; sudo make go-tidy-all`.

If you are going to be working with Golang a lot VSCode also has a solid Go plugin that auto formats everything (if you run into issues installing the modules, you might need to `chown -R $USER:$USER ~/go`).

**Review by dmcilvaney — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Comment by dallasd1** [comment]
> The go auto formatter looks unhappy, you can invoke all our tidy scripts via `cd ./toolkit; sudo make go-tidy-all`.
> 
> If you are going to be working with Golang a lot VSCode also has a solid Go plugin that auto formats everything (if you run into issues installing the modules, you might need to `chown -R $USER:$USER ~/go`).

Ah, thanks for pointing this out. The go-tidy-all command resolved some the formatting issues.

File: toolkit/tools/specreader/specreader.go [hunk 1]
```diff
@@ -82,17 +84,34 @@ func parseSPECsWrapper(buildDir, specsDir, rpmsDir, srpmsDir, distTag, outputFil
...
+				err := fmt.Errorf("Failed to parse cross specs (%w)", parseError)
+				return err
+			}
+		}
+
 		return parseError
 	}
 
 	if chroot != nil {
-		logger.Log.Info("Parsing SPECs inside a chroot environment")
```
[the remark this answers is not in the document]
Comment (dallasd1) [reply toolkit/tools/specreader/specreader.go]: This was not intended, logs added back

File: toolkit/docs/building/building.md [hunk 2]
```diff
@@ -688,6 +688,7 @@ To reproduce an ISO build, run the same make invocation as before, but set:
 | USE_PACKAGE_BUILD_CACHE       | y                                                                                                      | Skip building a package if it and its dependencies are already built.
 | NUM_OF_ANALYTICS_RESULTS      | 10                                                                                                     | The number of entries to print when using the `graphanalytics` tool. If set to 0 this will print all available results.
 | REBUILD_DEP_CHAINS            | y                                                                                                      | Rebuild packages if their dependencies need to be built, even though the package has already been built.
+| TARGET_ARCH            |                                                                                                       | The architecture of the machine that will run the package binaries.
```
Comment (dmcilvaney) [inline toolkit/docs/building/building.md:691]: nit: Spacing got a bit out of whack here.
```suggestion
| TARGET_ARCH                   |                                                                                                        | The architecture of the machine that will run the package binaries.
```

**Comment by dmcilvaney** [comment]
Got a valid build all the way through the pipelines (except ARM which is running into unrelated issues right now)

**Review by christopherco — APPROVED** [review]
:shipit:

**Pull request merged**